File: apps/passenger-uni-app/uni-app/src/api/ai.js
import { request } from '@/utils/request'

// AI 对话
export const chatAI = (data) =>
  request({ url: '/api/v1/ai/chat', method: 'POST', data, silent: true })

// 雨天路线规划
export const planRainRoute = (data) =>
  request({ url: '/api/v1/ai/rain-route', method: 'POST', data, silent: true })

// 带路线的 AI 对话
export const chatWithRoute = (data) =>
  request({ url: '/api/v1/ai/chat-with-route', method: 'POST', data, silent: true })

// 积水上报
export const submitFloodReport = (data) =>
  request({ url: '/api/v1/ai/flood-report', method: 'POST', data })

File: admin-platform/server/router/carpool/ai.go
package carpool

import (
	"ride-hailing/admin-server/middleware"

	"github.com/gin-gonic/gin"
)

type AIRouter struct{}

func (r *AIRouter) InitAIRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	aiRouter := Router.Group("carpool/ai").Use(middleware.OperationRecord())
	aiRouterWithoutRecord := Router.Group("carpool/ai")
	travelPublicRouter := PublicRouter.Group("travel")
	{
		aiRouter.POST("chat", aiApi.Chat)
		aiRouter.POST("rain-route", aiApi.PlanRainRoute)
		aiRouter.POST("chat-with-route", aiApi.ChatWithRainRoute)
		aiRouter.POST("flood-report", aiApi.ReportFlooding)
		aiRouter.POST("flood-report/audit", aiApi.AuditFloodReport)
		aiRouter.POST("export", aiApi.ExportAI)
	}
	{
		aiRouterWithoutRecord.GET("summary", aiApi.GetAISummary)
		aiRouterWithoutRecord.GET("conversation/list", aiApi.ListConversationLogs)
		aiRouterWithoutRecord.GET("route-plan/list", aiApi.ListRoutePlanLogs)
		aiRouterWithoutRecord.GET("flood-report/list", aiApi.ListFloodReports)
	}
	{
		travelPublicRouter.POST("route-info", aiApi.GetTravelRouteInfo)
	}
}
